### PR TITLE
Add support for request an MFA token code when required.

### PR DESCRIPTION
**Pull request by normj**
Check to see if the AWSCredententials is an assume role credentials and if so provide a callback to request the MFA code from the user.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/Amazon.Common.DotNetCli.Tools/AssumeRoleMfaTokenCodeCallback.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+using Amazon.Runtime;
+
+namespace Amazon.Common.DotNetCli.Tools
+{
+    /// <summary>
+    /// Handle the callback from AWSCredentials when an MFA token code is required.
+    /// </summary>
+    internal class AssumeRoleMfaTokenCodeCallback 
+    {
+        AssumeRoleAWSCredentialsOptions Options { get; set; }
```
Comment (vellozzi) [inline src/Amazon.Common.DotNetCli.Tools/AssumeRoleMfaTokenCodeCallback.cs]: Could be private.  Doesn't need to be/shouldn't be referenced externally.

File: src/Amazon.Common.DotNetCli.Tools/Utilities.cs [hunk 2]
```diff
@@ -694,5 +694,33 @@ public static ExecuteShellCommandResult ExecuteShellCommand(string workingDirect
                 return new ExecuteShellCommandResult(proc.ExitCode, capturedOutput.ToString());
             }            
         }
+
+        public static string ReadSecretFromConsole()
```
Comment (boblodgett) [inline src/Amazon.Common.DotNetCli.Tools/Utilities.cs]: Is the current behavior of accepting ESC and TAB characters valid? Also, is setting the key to empty the desired effect for just pressing enter or should it check for this case?

File: src/Amazon.Common.DotNetCli.Tools/AssumeRoleMfaTokenCodeCallback.cs (hunk 1 above)
Comment (normj) [reply src/Amazon.Common.DotNetCli.Tools/AssumeRoleMfaTokenCodeCallback.cs]: Updated. Keep in mind though the default is private when a visibility modifier is not specified

File: src/Amazon.Common.DotNetCli.Tools/Utilities.cs (hunk 2 above)
Comment (normj) [reply src/Amazon.Common.DotNetCli.Tools/Utilities.cs]: Updated code to skip the initial ASCII control characters like ESC and TAB

File: src/Amazon.Common.DotNetCli.Tools/AssumeRoleMfaTokenCodeCallback.cs (hunk 1 above)
Comment (vellozzi) [reply src/Amazon.Common.DotNetCli.Tools/AssumeRoleMfaTokenCodeCallback.cs]: Are you sure?  I thought it was internal.
I prefer it to be explicit anyway.

**Pull request merged**